tests/test_ghostty_zsh_job_table_saturation_guard.py

```
@@ -0,0 +1,229 @@
+            try:
+                proc.wait(timeout=5)
+            except subprocess.TimeoutExpired:
+                proc.kill()
+            os.close(master)
+
+    return bytes(output)
+
+
+def _capture_status_marker_session(env: dict[str, str], zsh_path: str) -> bytes:
+    master, slave = pty.openpty()
+    proc = subprocess.Popen(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)